app/controllers/users_controller.rb

```diff
@@ -0,0 +1,72 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :edit, :update, :destroy]
+
+  def index
+    params[:inicial_date] = Date.today.beginning_of_month unless params[:inicial_date]
+    params[:final_date] = Date.today.end_of_month unless params[:final_date]
+    @users = User.all
+    @users = @users.filter_by_name(params[:name_search]) if params[:name_search]
+    inicial_date = params[:inicial_date].to_date rescue nil
+    final_date = params[:final_date].to_date rescue nil
+    @users = @users.filter_by_initial_date(inicial_date) if inicial_date
+    @users = @users.filter_by_final_date(final_date) if final_date
```

Comment:
We could refactor this method to make it smaller. Somthing like this:

```
def index
    initial_date = do_your_magic_trick_to_define_the_initial_date
    final_date = do_your_magic_trick_again_for_final_date
    name_search = do_it_again_for_name_search_params
    @users = if name_search.present?
                       User.scoped_by_name
                     elsif initial_date.present?
                        User.scoped_by_initial_date
                     elsif final_date.present?
                         User.scoped_by_final_date
                     else
                         User.unscoped
                     end
```

The above example is a way to refactor the method, but it can be further improved. And another thing to keep in mind:
If the user passes a name, an initial date and a final date, what will happen? :shipit: What would be the result of the search? It will be what the user expect? 🍩